sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -774,9 +844,15 @@ protected Symbol visitSubqueryExpression(SubqueryExpression node, ExpressionAnal
              * But there are no other row-expressions yet. In addition the cast functions and operators don't work with
              * row types (yet).
              *
-             * Since we only support 1 column and only single-row subselects it is okay to use the inner type directly.
+             * However, we support a single column RowType through the SingleColumnTableType.
              */
-            return new SelectSymbol(relation, fields.get(0).valueType());
+            Field field = fields.get(0);
+            SingleColumnTableType singleColumnTableType = new SingleColumnTableType(field.valueType());
+            SelectSymbol selectSymbol = new SelectSymbol(relation, singleColumnTableType);
+            return context.allocateFunction(
+                getBuiltinFunctionInfo(SingleValueFunction.NAME, Collections.singletonList(singleColumnTableType)),
+                // needs to be a mutable list...
```

Comment:
Crate manipulates symbols in-place. This doesn't work if we use `Collections.singletonList(..)` here. Added a note.